### DISCUSSION TITLE
Why my script cannot show the spinner in dialog.

### DISCUSSION THREAD

**Discussion by italanchan** [zauberzeug/nicegui#1394] [Q&A]
As the search function is long run over 10s, it is need to display waiting message or symbol under progress.  But my script cannot show the spinner in dialog.  How to fix?

The code:

```py
from nicegui import ui
from time import sleep

def search():
    with ui.dialog():
        sp = ui.spinner('dots', size='xl')
        sp.visible = True
        sleep(5) # doing long run tasks over 10s ....

ui.button('Search', on_click=search)

ui.run()
```

**Comment by falkoschindler** [comment]
Hi @italanchan, have a look into https://github.com/zauberzeug/nicegui/discussions/695 and https://github.com/zauberzeug/nicegui/discussions/816. I'll close this question as duplicate.

**Comment by italanchan** [comment]
Sorry, I don't understand #695 and #816.  May be my question is poor.

In fact, my case have no use sleep function, just the function is running over 10s.  As the user reported the clicked seem the app is no any response, in fact, the program is running but the UI cannot response after clicked.  However, user would click again and again, it make the program is hang.  

Or, 

Is there any method or simple way to disable button when clicked, and enable after the task is completed ?

**Reply by rodja** [reply]
You need to use asyncio. Please have a look at our FAQ: https://github.com/zauberzeug/nicegui/wiki/FAQs#how-to-not-get-the-reconnect-popup-or-avoid-laggy-interaction.
